ui/src/BaseEditor.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { DialogResponses } from './DialogResponses' ;
+import { UserCancelledError } from './errors';
+import { localize } from "./localize";
+import { TemporaryFile } from './utils/TemporaryFile';
+
+// tslint:disable-next-line:no-unsafe-any
+export abstract class BaseEditor<ContextT> implements vscode.Disposable {
+    private fileMap: { [key: string]: [vscode.TextDocument, ContextT] } = {};
+    private ignoreSave: boolean = false;
+
+    constructor(private readonly showSavePromptKey: string, private readonly outputChannel?: vscode.OutputChannel) {
+    }
+
+    public abstract getData(context: ContextT): Promise<string>;
+    public abstract updateData(context: ContextT, data: string): Promise<string>;
+    public abstract getFilename(context: ContextT): Promise<string>;
+    public abstract getSize(context: ContextT): Promise<number>;
+    public abstract getSaveConfirmationText(context: ContextT): Promise<string>;
+
+    public async showEditor(context: ContextT, sizeLimit?: number /* in Megabytes */): Promise<void> {
+        const fileName: string = await this.getFilename(context);
+
+        this.appendToOutput(localize('opening', 'Opening "{0}" ...', fileName));
+        if (sizeLimit) {
+            const size: number = await this.getSize(context);
+            if (size > sizeLimit) {
+                const message: string = localize('tooLargeError', '"{0}" is too large to download.', fileName);
+                this.appendLineToOutput(localize('failed', " Failed."));
+                throw new Error(message);
+            }
+        }  else {
+            try {
+                const localFilePath: string = await TemporaryFile.create(fileName);
+                const document: vscode.TextDocument = await vscode.workspace.openTextDocument(localFilePath);
+                this.fileMap[localFilePath] = [document, context];
+                const data: string = await this.getData(context);
+                const textEditor: vscode.TextEditor = await vscode.window.showTextDocument(document);
+                await this.updateEditor(data, textEditor);
+                this.appendLineToOutput(' Done.');
+            } catch (error) {

```

Comment:
Just remove this entire 'catch' block and let the error propagate. Our extensions have more robust error parsing/handling already